src/basecamp.cpp

```diff
@@ -3,17 +3,62 @@
 #include "output.h"
 #include "string_formatter.h"
 #include "translations.h"
+#include "enums.h"
+#include "item_group.h"
+#include "map.h"
+#include "map_iterator.h"
+#include "mapbuffer.h"
+#include "mapdata.h"
+#include "overmap.h"
+#include "overmap_ui.h"
+#include "overmapbuffer.h"
+#include "npc.h"
+#include "recipe.h"
+#include "recipe_groups.h"
+#include "requirements.h"
+#include "skill.h"
+
+#include "faction_camp.h"
 
 #include <algorithm>
+#include <array>
 #include <sstream>
+#include <map>
+#include <string>
+#include <vector>
+
+static const std::string base_dir = "[B]";
+static bool valid_dir( const std::string &dir )
+{
+    if( dir == base_dir || dir == "[N]" || dir == "[S]" || dir == "[E]" || dir == "[W]" ||
+        dir == "[NE]" || dir == "[NW]" || dir == "[SE]" || dir == "[SW]" ) {
+        return true;
+    }
+    return false;
+}
+
+static inline const std::string faction_encode( const std::string &type )
```

Comment:
ìnline` *and* `static` on a free function are almost never needed. In this case the `inline`does not add anything useful. (Compilers will decide on their own which function to inline. `inline` is merely a hint, but not a requirement.)

Reply:
Fixed, thanks.